src/norm/pool.nim

```diff
@@ -0,0 +1,106 @@
+import std/locks
+
+import sqlite, postgres
+
+
+type 
+  DbConn = sqlite.DbConn | postgres.DbConn
+  Pool*[T: DbConn] = ref object 
+    defaultSize: Natural
+    conns: seq[T]
+    poolExhaustedPolicy: PoolExhaustedPolicy
+    lock: Lock
+  PoolExhaustedError* = object of CatchableError
+  PoolExhaustedPolicy* = enum
+    pepRaise
+    pepExtend
+
+
+func newPool*[T: DbConn](defaultSize: Positive, poolExhaustedPolicy = pepRaise): Pool[T] =
+  ##[ Create a connection pool of the given size.
+
+  ``poolExhaustedPolicy`` defines how the pool reacts when a connection is requested but the pool has no connection available:
+
+    - ``pepRaise`` (default) means throw ``PoolExhaustedError``
+    - ``pepExtend`` means throw “add another connection to the pool.”
+  ]##
+
+  result = Pool[T](defaultSize: defaultSize, conns: newSeq[T](defaultSize), poolExhaustedPolicy: poolExhaustedPolicy)
+
+  initLock(result.lock)
+
+  for conn in result.conns.mitems:
+    conn =
+      when T is sqlite.DbConn:
+        sqlite.getDb()
+      elif T is postgres.DbConn:
+        postgres.getDb()
+
+func defaultSize*(pool: Pool): Natural =
+  pool.defaultSize
+
+func size*(pool: Pool): Natural =
+  len(pool.conns)
+
+func pop*[T: DbConn](pool: var Pool[T]): T =
```

Comment:
A log of level debug could be useful here to log the that after taking out/before taking out there's X connections. Though arguably the user could also do this themselves if needed before/after calling `pop`. Same for `add` and maybe `reset`